src/historical.ts

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import { MatcherType, SupportedMatcher } from "./types/enums";
+import { Context } from "./types/engine";
+
+export function checkForHistoricalMatcher(
+  eventCount: any,
+  matcherKey: SupportedMatcher,
+  value: number
+) {
+  switch (matcherKey) {
+    case MatcherType.GREATER_THAN:
+      return eventCount > value;
+    case MatcherType.GREATER_THAN_OR_EQUAL_TO:
+      return eventCount >= value;
+    case MatcherType.LESS_THAN:
+      return eventCount < value;
+    case MatcherType.LESS_THAN_OR_EQUAL_TO:
+      return eventCount <= value;
+    case MatcherType.EQUALS:
+      return eventCount === value;
+    case MatcherType.NOT_EQUALS:
+      return eventCount !== value;
+    default:
+      return false;
+  }
+}
+
+/**
+ * Iterates through the rule events and checks if any event occurs in context between
+ * the from and to timestamps defined in rule. It returns 1 (true) if the event is in the context and 0 otherwise.
+ *
+ * @param {Array<any>} events The list of events to query.
+ * @param {Context} context The context object.
+ * @param {number} from The from timestamp.
+ * @param {number} to The to timestamp.
+ * @returns {number} 1 if the event is in the context and 0 otherwise.
+ */
+export function queryAndCountAnyEvent(
+  events: Array<any>,
+  context: Context,
+  from: number,
+  to: number
+) {
+  for (let i = 0; i < events.length; i += 1) {
```

Comment:
I recommend using Array.forEach or Array.reduce where possible instead of for loops like this.  It's the [For in loop](https://www.w3schools.com/js/js_loop_forin.asp) that we can't use.